### PR TITLE
macros: allow attribute invocations at the crate root

### LINKED ISSUE #36617: Module level attributes are being parsed before plugins are registered
Previously I was using a plugin to register a `MultiModifier`, and the plugin users would use a `#![modifier]` attribute in the entry module of their programs, so the plugin could rewrite anything inside the module.

But on the latest nightly, this results in a compilation error: 
```
error: The attribute `modifier` is currently unknown to the compiler and may have meaning added to it in the future (see issue #29642)
```
This leads me to believe that module level attributes are now being parsed before the plugins finish registering.

Tagging @jseyfried.

### PR DESCRIPTION
**Pull request by jseyfried**
Fixes #36617.
r? @nrc

### CODE REVIEW

**Comment by jseyfried** [comment]
I unintentionally disallowed attribute macro invocations at the crate root in #36214.

File: src/test/compile-fail/issue-36617.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![derive(Copy)] //~ ERROR `derive` may only be applied to structs and enums
```
Comment (nrc) [inline src/test/compile-fail/issue-36617.rs]: should be ``derive` may only be applied to structs, enums and unions`

**Comment by nrc** [comment]
r=me with test fixed

File: src/test/compile-fail/issue-36617.rs (hunk 1 above)
Comment (jseyfried) [inline src/test/compile-fail/issue-36617.rs]: Thanks

**Pull request merged**